libcodechecker/logger.py

```diff
@@ -54,137 +43,126 @@ def debug_analyzer(self, msg, *args, **kwargs):
-        format_orig = self._fmt
-
-        # Get a format based on the most verbose loglevel set by the user.
-        format_strings = CustomFormatter.formatter_map[self.level]
-
-        # Replace the original format.
-        if record.levelno == logging.DEBUG:
-            self._fmt = format_strings['debug_fmt']
-        if record.levelno == logging.DEBUG_ANALYZER:
-            self._fmt = format_strings['debug_fmt']
-        elif record.levelno == logging.INFO:
-            self._fmt = format_strings['info_fmt']
-        elif record.levelno == logging.ERROR:
-            self._fmt = format_strings['error_fmt']
-        elif record.levelno == logging.WARNING:
-            self._fmt = format_strings['error_fmt']
-        elif record.levelno == logging.CRITICAL:
-            self._fmt = format_strings['error_fmt']
+def add_verbose_arguments(parser):
+    """
+    Verbosity level arguments.
+    """
+    parser.add_argument('--verbose', type=str, dest='verbose',
+                        choices=CMDLINE_LOG_LEVELS,
+                        default='info',
+                        help='Set verbosity level.')
 
-        result = logging.Formatter.format(self, record)
 
-        self._fmt = format_orig
+def get_logger(name):
+    """
+    Return a loggger instance if already exists with the given name.
+    """
+    return logging.getLogger(name)
 
-        return result
 
+def validate_loglvl(log_level):
+    """
+    Should return a valid log level name
+    """
+    log_level = log_level.upper()
+    if log_level not in map(string.upper, CMDLINE_LOG_LEVELS):
+        return "INFO"
+    else:
+        return log_level
 
-class LoggerFactory(object):
-    log_level = logging.INFO
-    loggers = []
 
-    short_format_handler = logging.StreamHandler(stream=sys.stdout)
-    long_format_handler = logging.StreamHandler(stream=sys.stdout)
+class LOG_CFG_SERVER(object):
+    """
+    Initialize a log configuration server for dynamic log configuration.
+    The log config server will only be started if the
+    'CC_LOG_CONFIG_PORT' environment variable is set.
+    """
 
-    custom_formatter = CustomFormatter()
+    def __init__(self, log_level='INFO'):
 
-    short_format_handler.setFormatter(custom_formatter)
-    long_format_handler.setFormatter(custom_formatter)
+        # Configure the logging with the default config.
+        setup_logger(log_level)
 
-    handlers = {
-        logging.INFO: short_format_handler,
-        logging.DEBUG: long_format_handler,
-        logging.DEBUG_ANALYZER: short_format_handler}
+        self.log_server = None
 
-    @classmethod
-    def get_log_level(cls):
-        return cls.log_level
+        log_cfg_port = os.environ.get('CC_LOG_CONFIG_PORT')
 
-    @classmethod
-    def set_log_level(cls, level):
-        for logger in cls.loggers:
-            logger.removeHandler(cls.handlers[LoggerFactory.log_level])
+        if log_cfg_port:
+            self.log_server = logging.config.listen(int(log_cfg_port))
```

Comment:
What happens if the listen server gets a log configuration that is invalid, or when someone just simply puts data into the socket?

Reply:
Running a log server is a security risk as noted [here](https://docs.python.org/2/library/logging.config.html#logging.config.listen) that is why it does not start by default only when configured. It does not accept connections from remote machines so a new configuration can be loaded only from localhost. I've put a minimal json validation into the sender script as a check before sending the config, but I know everything can be sent to that port which might cause problems.